engines/opensearch/OpenSearchCollection.py

```diff
@@ -0,0 +1,192 @@
+import random
+from re import search
+import requests
+from engines.Collection import Collection
+from engines.opensearch.config import OPENSEARCH_URL
+import time
+import json
+from pyspark.sql import Row, SparkSession
+import numbers
+
+def generate_vector_search_request(search_args):
+    vector = search_args.pop("query")
+    query_fields = search_args.pop("query_fields", [])
+    if not isinstance(query_fields, list):
+        raise TypeError("query_fields must be a list")
+    elif len(query_fields) == 0:
+        raise ValueError("You must specificy at least one field in query_fields")
+    else: 
+        field = query_fields[0]
+    k = search_args.pop("k", 10)
+    if "limit" in search_args: 
+        if int(search_args["limit"]) > k:
+            k = int(search_args["limit"]) #otherwise will only get k results
+    request = {"query": {"knn": {field: {"vector": vector,
+                                            "k": k}}},
+               "size": search_args.get("limit", 5)}             
+    if "log" in search_args:
+        print("Search Request:")
+        print(json.dumps(request, indent="  "))
+    return request
+        
+def create_filter(field, value):
+    if value != "*":
+        key = "terms" if isinstance(value, list) else "term"
+        return {key: {field: value}}
+    else:
+        return {"exists": {"field": field}}
+    
+def query_string_clause(search_args):
+    "Returns the query string from the args or from concatenating query clause strings"
+
+    def retrive_strings(c): return c if isinstance(c, str) else ""
+
+    query_string = "*"
+    if "query" in search_args:
+        if isinstance(search_args["query"], list):
+            query_string = " ".join(list(map(retrive_strings, search_args["query"])))
+        else:
+            query_string = search_args["query"]
+    return query_string.strip()
+
+def should_clauses(search_args):
+    return list(filter(lambda c: isinstance(c, dict) and "geo_distance" not in c,
+                       search_args.get("query", [])))
+
+def must_clauses(search_args):
+    return list(filter(lambda c: isinstance(c, dict) and "geo_distance" in c,
+                       search_args.get("query", [])))
+
+def is_vector_search(search_args):
+    return "query" in search_args and \
+           isinstance(search_args["query"], list) and \
+           len(search_args["query"]) == len(list(filter(lambda o: isinstance(o, numbers.Number),
+                                                        search_args["query"])))
+
+class OpenSearchCollection(Collection):
+    def __init__(self, name, id_field="_id"):
+        super().__init__(name)
+        self.id_field = id_field
+        
+    def commit(self):
```

Comment:
In OpenSearch, the commit is done by the "flush" operation. https://opensearch.org/docs/2.18/api-reference/index-apis/flush/

Is this not needed? I'd assume we'll have consistency issues between cells without this, no?